receiver/k8sobjectsreceiver/receiver.go

```diff
@@ -141,37 +144,69 @@ func (kr *k8sobjectsreceiver) startPull(ctx context.Context, config *K8sObjectsC
 }
 
 func (kr *k8sobjectsreceiver) startWatch(ctx context.Context, config *K8sObjectsConfig, resource dynamic.ResourceInterface) {
-
 	stopperChan := make(chan struct{})
 	kr.mu.Lock()
 	kr.stopperChanList = append(kr.stopperChanList, stopperChan)
 	kr.mu.Unlock()
 
-	resourceVersion, err := getResourceVersion(ctx, config, resource)
-	if err != nil {
-		kr.setting.Logger.Error("could not retrieve an initial resourceVersion", zap.String("resource", config.gvr.String()), zap.Error(err))
-		return
-	}
 	watchFunc := func(options metav1.ListOptions) (apiWatch.Interface, error) {
 		options.FieldSelector = config.FieldSelector
 		options.LabelSelector = config.LabelSelector
 		return resource.Watch(ctx, options)
 	}
 
+	_, cancel := context.WithCancel(ctx)
```

Comment:
I don't think `WithCancel` mutates the original context. So calling `cancel` doesn't affect it, it's only applied to the context that is returned as the first value, which is ignored

Reply:
Not mutating the original context is definitely what I was going for. I believe we only care about the cancel function for the context of breaking out of this function. Are you thinking I should pass the returned context into the subsequent functions instead of the original?

Reply:
Reading through the docs again you're right, we definitely want to use the new context since it will close if the original does or `cancel` is called.